api/v1beta1/ovndbcluster_types.go

```diff
@@ -197,7 +197,7 @@ func (instance OVNDBCluster) GetInternalEndpoint() (string, error) {
 }
 
 func (instance OVNDBCluster) GetExternalEndpoint() (string, error) {
-	if instance.Status.DBAddress == "" {
+	if instance.Spec.NetworkAttachment != "" && instance.Status.DBAddress == "" {
 		return "", fmt.Errorf("external DBEndpoint not ready yet for %s", instance.Spec.DBType)
```

Comment:
is it not ready *yet*, or will it not be ever available (with the spec without attachment)? Should these two branches then have separate error messages? What do you think?

Reply:
I think when DB Cluster have No NetworkAttachment we don't need to raise any kind of error as with it DBAddress will always be unset.

Reply:
If you don't return an error at all, then `deleteExternalConfigMaps` in line 461 will never be called to clean up obsolete external CM. Is it what you intend here?